### ISSUE TITLE
CalendarByPin endpoint returning 403 forbidden since 12:00 IST

### ISSUE THREAD

**Issue by rohitashwachaks** [cowinapi/developer.cowin#169]
- I have not exceeded the rate limit
- My app is hosted in India.
- I was getting 200 response codes before 12:00 noon

It has started receiving tons of 403 since 12. Any Idea why?

**Comment by ajinair** [comment]
Please leave atleast 5 secs gap for each request. else you will get 403

**Comment by rohitashwachaks** [comment]
Even if the total calls are under 100 calls per 5 minutes?

**Comment by ajinair** [comment]
Yaa thats what I saw when I reduced the interval to 3 seconds and multiple instance. They are not able to serve for the request id within that short time and ends up 403. try with min of 5 secs atleast. May be institutions with access to protected API have better response time.

**Comment by rohitashwachaks** [comment]
@ajinair it'll be difficult for me to try your method because I made my code asynchronous.
I'll have change the whole structure to introduce this...
.
However, I rechecked....
My code is working from my local device,
 not from Azure Cloud

**Comment by ipankajkumar93** [comment]
There is no such thing as waiting for 5 sec before each call. Don't misguide
We are sending array of pinCodes after 1 sec each, no problem at all

**Comment by kartikeyas00** [comment]
> There is no such thing as waiting for 5 sec before each call. Don't misguide
> We are sending array of pinCodes after 1 sec each, no problem at all

Wait how are you doing that? Isn't there a limit of 100 calls per 5 minutes?

**Comment by ajinair** [comment]
@ipankajsingh Hello mister you should send your request with a sleep of 5 secs atleast and not any such thing from Cowin API. I send array of districts itself and get schedule for next 14 days. with a gap of 5 secs for each call.

**Comment by ipankajkumar93** [comment]
You do it because you decided to do it.
There is only 1 restrcition from co-WIN, 100 per per 5 min
Whether sumone consumes those 100 res in 1 min or 5 min is upto them

**Comment by reyasmohammed** [comment]
> 
> 
> You do it because you decided to do it.
> There is only 1 restrcition from co-WIN, 100 per per 5 min
> Whether sumone consumes those 100 res in 1 min or 5 min is upto them

for me it was working fine yesterday from Azure cloud (India location). But from today early morning, it stopped giving results.
I am giving a try with some seconds delay now.. In my case I have only 7 calls. that too with 30 minutes delay/. 
Now added delay between these 7 calls too

**Comment by kartikeyas00** [comment]
> Whether sumone consumes those 100 res in 1 min or 5 min is upto them

Makes sense!

**Comment by ajinair** [comment]
Not sure if an API can return results every 1 sec behind which you are sending another.

**Comment by rohitashwachaks** [comment]
API's are usually asynchronous and scalable. That shouldn't be a problem.

@reyasmohammed I'm facing the some issue!!
I literally migrated from USCentral to India to get around the foreign IP Blacklisting problem.
Now even the IndiaCentral is being blocked. FFS
.
It's working from local but not from Cloud

**Comment by reyasmohammed** [comment]
> 
> 
> API's are usually asynchronous and scalable. That shouldn't be a problem.
> 
> @reyasmohammed I'm facing the some issue!!
> I literally migrated from USCentral to India to get around the foreign IP Blacklisting problem.
> Now even the IndiaCentral is being blocked. FFS
> .
> It's working from local but not from Cloud

Yeah, same for me in Central India.. Now moved to South India.. using Azure Functions..

**Comment by rohitashwachaks** [comment]
> > API's are usually asynchronous and scalable. That shouldn't be a problem.
> > @reyasmohammed I'm facing the some issue!!
> > I literally migrated from USCentral to India to get around the foreign IP Blacklisting problem.
> > Now even the IndiaCentral is being blocked. FFS
> > .
> > It's working from local but not from Cloud
> 
> Yeah, same for me in Central India.. Now moved to South India.. using Azure Functions..

Is it working in South India?
I'm Using Azure Functions Too XD

**Comment by reyasmohammed** [comment]
> 
> 
> > > API's are usually asynchronous and scalable. That shouldn't be a problem.
> > > @reyasmohammed I'm facing the some issue!!
> > > I literally migrated from USCentral to India to get around the foreign IP Blacklisting problem.
> > > Now even the IndiaCentral is being blocked. FFS
> > > .
> > > It's working from local but not from Cloud
> > 
> > 
> > Yeah, same for me in Central India.. Now moved to South India.. using Azure Functions..
> 
> Is it working in South India?
> I'm Using Azure Functions Too XD

South India working as of now.. dont know how long...

**Comment by rohitashwachaks** [comment]
@reyasmohammed Are you on the consumption plan?
I'm unable to deploy to India South or west.
Not sure if it's a free trial restriction or if I'm doing something wrong

**Comment by reyasmohammed** [comment]
> 
> 
> @reyasmohammed Are you on the consumption plan?
> I'm unable to deploy to India South or west.
> Not sure if it's a free trial restriction or if I'm doing something wrong

I am not using free trial, may be there is restriction in trial.
